### DISCUSSION TITLE
Different ID first time load and second time load Safari

### DISCUSSION THREAD

**Discussion by geodan4ik** [fingerprintjs/fingerprintjs#765] [Q&A]
Hey guys! I am using the standard fingerprintjs code 

```js
<script>
    // Initialize the agent at application startup.
    const fpPromise = import('https://openfpcdn.io/fingerprintjs/v3')
      .then(FingerprintJS => FingerprintJS.load())
  
    // Get the visitor identifier when you need it.
    fpPromise
      .then(fp => fp.get())
      .then(result => {
        // This is the visitor identifier:
        const visitorId = result.visitorId
        alert(visitorId)
      })
  </script>
```

It works fine in Firefox, but in Safari, the first time load after history reset shows one ID and then after refresh of the page it shows another ID. This happens only on computer, mobile safari works well. 

First load after history reset: b66307e181851ec379e25bb825ced78e
Second load after history reset: 2acaabbd7a1b3930afc5b0aaba349846
Third load after history reset: 2acaabbd7a1b3930afc5b0aaba349846

What should I do?
And is there a way of deleting the fingerprint js on a button click. for example a user presses on this button and now he or she has a new ID

**Comment by makma** [comment]
Hey @geodan4ik,
What do you mean by history reset?
Please, share entropy sources for both visitorIds with us (e.g. from [here](https://fingerprintjs.github.io/fingerprintjs/)).

> And is there a way of deleting the fingerprint js on a button click. for example a user presses on this button and now he or she has a new ID.

VisitorId is not stored, it's calculated, therefore it's not possible to delete it.

**Reply by geodan4ik** [reply]
Sorry, i might have written the question a bit in a weird form 
I dont know how to get the visitors ID as like shown in the example, i can only see the ID that shows on console or as an alert. 
I can alternatively describe better the chronology of my action:
1) open my website with the fingerprint js 
2) the alert shows me the unique ID 
3) i delete the history of the browser 
4) open the same page, alert shows me unique ID (always the same after history is deleted) 
5) when i reload the page, the unique ID changes (always the same when the webpage is loaded second time and identical to the alert on the step  "2)" )

**Reply by makma** [reply]
Hello @geodan4ik,
We have reproduced the issue on our side, thank you for letting us know. We don't know why Safari interprets data in a different way randomly, also we weren't able to stabilize this source. However, this is quite a rare use case and two generated fingerprints are stable and they don't change in time.

**Reply by Valve** [reply]
should we create an issue @makma we need to do something about this Safari behavior.
Maybe we'll get a contribution to fix it, who knows.

**Reply by makma** [reply]
@Valve, issue created - https://github.com/fingerprintjs/fingerprintjs/issues/768
